pkg/test/expectations/expectations.go

```
@@ -302,15 +302,15 @@ func ExpectProvisionedNoBindingWithOffset(offset int, ctx context.Context, c cli
 	return bindings
 }
 
-func ExpectMachineDeployedNoNode(ctx context.Context, c client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider, m *v1alpha5.Machine) *v1alpha5.Machine {
+func ExpectMachineDeployedNoNode(ctx context.Context, c client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider, m *v1alpha5.Machine) (*v1alpha5.Machine, error) {
 	return ExpectMachineDeployedNoNodeWithOffset(1, ctx, c, cluster, cloudProvider, m)
 }
 
-func ExpectMachineDeployedNoNodeWithOffset(offset int, ctx context.Context, c client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider, m *v1alpha5.Machine) *v1alpha5.Machine {
+func ExpectMachineDeployedNoNodeWithOffset(offset int, ctx context.Context, c client.Client, cluster *state.Cluster, cloudProvider cloudprovider.CloudProvider, m *v1alpha5.Machine) (*v1alpha5.Machine, error) {
 	resolved, err := cloudProvider.Create(ctx, m)
 	// TODO @joinnis: Check this error rather than swallowing it. This is swallowed right now due to how we are doing some testing in the cloudprovider
 	if err != nil {
-		return nil
```

Just curious why we can't do a `Expect(err).ToNot(BeNil())` here and handle everything thusly. I suppose the comment above is a hint why we aren't able to do that?

---

Yeah, I think we want to get to the point that we can. But right now we have some tests that are relying on the scheduling to fail while ignoring the error

---

I definitely think there is room for improvement here